llvm/lib/Target/Mips/MipsExpandPseudo.cpp

```diff
@@ -473,48 +476,38 @@ bool MipsExpandPseudo::expandAtomicBinOpSubword(
     unsigned SELOldVal = IsMax ? SELEQZ : SELNEZ;
     unsigned MOVIncr = IsMax ? MOVN : MOVZ;
 
-    // For little endian we need to clear uninterested bits.
-    if (STI->isLittle()) {
-      if (!IsUnsigned) {
-        BuildMI(loopMBB, DL, TII->get(Mips::SRAV), OldVal)
-            .addReg(OldVal)
-            .addReg(ShiftAmnt);
-        BuildMI(loopMBB, DL, TII->get(Mips::SRAV), Incr)
-            .addReg(Incr)
```

Comment:
For `RegState::Kill`, I noticed that it is also used when dest reg is same with one of src regs, not only in MIPS code.
In fact I am not sure about its really meanings.

`Incr` is an argument of this function passed by register $5. We can assume that it has been well sign-/zero- extended.

Reply:
@wzssyqa Do I need to remove the ```RegState::Kill``` of ```StoreVal```? The current test has not found any problems related to this. But I once encountered a problem related to ```RegState::Kill``` when I create a new reg like ```Scratch``` in function ```MachineBasicBlock *MipsTargetLowering::emitAtomicBinaryPartword``` and use it in function ```bool MipsExpandPseudo::expandAtomicBinOpSubword``` to replace ```OldVal```, and when the code is executed to ```slt```, an error is reported.

Reply:
It seems it's safe to use `RegState::Kill` for something like `a <- op(a)`.
I guess it is even more friendly to register allocator, Since the register allocator can even use different hardware register for before/after this `RegState::Kill` note.

Reply:
Yes, now it is OK for ```a <- op(a)```, the scenario where my lab error reported is ```b <- op(a, RegState::Kill) + op(a)```.